qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py

```diff
@@ -636,13 +636,15 @@ def from_sparse_list(obj, num_qubits):
         Args:
             obj (Iterable[Tuple[str, List[int], complex]]): The list 3-tuples specifying the Paulis.
             num_qubits (int): The number of qubits of the operator.
+            do_checks (bool): The flag of checking if it is already assigned (not I).
```

Comment:
I think `I` should be checked too, otherwise it's unclear how we interpret using an index multiple times 🙂 Could you add that?

Reply:
It is true that there is no reason to pass `I` with sparse format, so I don't think we need to allow `I`.

Reply:
Yes that's true, strictly speaking there's not reason to pass `I`. But it might be restrictive to explicitly forbid it because (1) if a user happens to have strings with `I` in them they would have to clean the string, which can be cumbersome (maybe the string is automatically constructed) and (2) we would have to add a check whether `I` is included. I think it would be easier for both users and us if we just allow any Pauli string. 

We could of course clean the input strings and only store the non-`I` ones, that might be good though!

Reply:
Then I feel that it is ok to not check `I` for the first version. Personally, I am concerned that it would complicate the implementation. As long as we make errors as long as the output is clearly strange (as issue pointed out), we don't have problems.

Reply:
I don't like neither because of the extra computational cost, but I'd prefer 1 using `set`.

Reply:
Thank you. I fixed the check condition. Could you review it?